File: app/src/main/java/net/squanchy/notification/NotificationInjector.java
package net.squanchy.notification;

import android.app.Service;

import net.squanchy.injection.ApplicationInjector;
import net.squanchy.injection.ServiceContextModule;

final class NotificationInjector {

    private NotificationInjector() {
        // no instances
    }

    public static NotificationComponent obtain(Service service) {
        return DaggerNotificationComponent.builder()
                .applicationComponent(ApplicationInjector.obtain(service))
                .serviceContextModule( new ServiceContextModule(service))
                .build();
    }
}

File: app/src/main/java/net/squanchy/notification/NotificationInjector.kt
package net.squanchy.notification

import android.app.Service

import net.squanchy.injection.ApplicationInjector
import net.squanchy.injection.ServiceContextModule

internal object NotificationInjector {

    fun obtain(service: Service): NotificationComponent {
        return DaggerNotificationComponent.builder()
                .applicationComponent(ApplicationInjector.obtain(service))
                .serviceContextModule(ServiceContextModule(service))
                .build()
    }
}
